docs/en/Community-Articles/2024-05-07-using-fluent-validation-with-abp/post.md

```diff
@@ -0,0 +1,95 @@
+Fluent Validation allows you to define your validation rules in a clear and flexible way. This means you can comfortably handle complex validation scenarios in your code. This makes your development process much more manageable. The readability that Fluent Validation offers really makes things easier. Having a clear understanding of what your validation rules do is a huge advantage when working on your code. In short, your code is cleaner and clearer. Fluent Validation is also very functional in terms of testability. By defining your validation rules in separate classes, you can easily test and maintain these rules. Fluent Validation is a widely used validation library on the .NET platform. As such, it has become a common standard among .NET developers. This provides advantages in terms of community support and compatibility. So, using Fluent Validation simplifies your development process by making your code more understandable, manageable and testable.
+
+## How to Use Fluent Validation in ABP?
+![fluent validation](./images/abp-fluentvalidation.png)
+
+Using Fluent validation with Abp is quite simple. Open a command line window in the folder of the project (.csproj file) and type the following command: 
+abp add-package Volo.Abp.FluentValidation
+If you have not created your abp project, review the [steps to create](https://docs.abp.io/en/abp/latest/Tutorials/Todo/Overall) it now.
+
+Create your product object like this: 
+````csharp
+  public class Product: FullAuditedAggregateRoot<Guid>
+   {
+       public string Name { get; set; } = string.Empty;
+       public decimal Price { get; set; }
+       public int Stock { get; set; }
+       public string? LicenseKey { get; set; }
+    }
+````
+Create the controller of your product object by following the [documentation](https://docs.abp.io/en/abp/latest/Tutorials/Todo/Index?UI=NG&DB=EF) 
+You must have ProductCreateDto :
+````csharp
+      public class ProductCreateDto
+    {
+        public string Name { get; set; } 
+        public decimal Price { get; set; }
+        public int Stock { get; set; }
+        public string? LicenseKey { get; set; }
+     }
+````
+Create the ProductCreateDtoValidator class in the products folder in the Application.Contacts layer. 
+````csharp
+  public class ProductCreateDtoValidator :AbstractValidator<ProductCreateDto>
+{
+    public ProductCreateDtoValidator()
+    {
+        RuleFor(p=>p.Name).
+          NotEmpty().
+            WithMessage("Product name cannot be empty");
+        
+        RuleFor(p=>p.Name).
+            MinimumLength(3).
+            MaximumLength(100).
+            WithMessage("Product name must be between 3 and 100 characters");
+
+        RuleFor(p => p.Stock).
+            GreaterThanOrEqualTo(0).
+            WithMessage("Product stock should not be negative");
+
+        RuleFor(p => p.Price).
+           GreaterThanOrEqualTo(0).
+           WithMessage("Product Price should not be negative");
+    }
+}
+````
```

Comment:
This is the main code block for this article. Therefore, I think you should explain this code block in plain and short sentences. 

For example, you should say that you can add a NotEmpty check or set min-max length for a property etc. (or specify a message that will be thrown when an validation is failed)

Reply:
I have added these paragraphs:

The validator class you create should inherit from `AbstractValidator`.  You should give `AbstractValidator` the class you want to validate generically. You have to create a constructor method. You must write the code in this constructor method. Fluent validation provides you with ready-made methods that you can use to write validations very easily. 

The **RuleFor** method allows you to write a validation rule. You must specify in the parameter for which property you want to write a validation rule. 

With the **NotEmpty** method you can specify that a property cannot be null.

With **MinimumLength** and **MaximumLength** you can specify the minimum and maximum number of characters the property can take.

With the **GreaterThanOrEqualTo** method you can specify that the value of the property must be greater than or equal to the value entered as a parameter.

**WithMessage** method you can specify the message to be sent when the validation fails.